utPlayer.py

```diff
@@ -31,42 +31,64 @@
 
 from framework.core.logModule import logModule
 from framework.plugins.ut_raft.interactiveShell import InteractiveShell
+from framework.plugins.ut_raft.configRead import ConfigRead
+
 class utPlayer():
     """
     UT Player class
     """
-    def __init__(self, session:object, playerTool:dict={'tool': 'gstreamer', 'prerequisites':[]}, log:logModule=None):
+    playerConfig = os.path.join(dir_path, "uPlayerConfig.yml")
+
+    def __init__(self, session:object, platform:str, player:str = "gstreamer", log:logModule=None):
         """
         Initializes player class.
 
         Args:
             session (class): The session object to communicate with the device
+            platform (str): Platfrom name
+            player (str, optional) : Player to use. Defaults to gstreamer
             log (class, optional): Parent log class. Defaults to None.
         """
         self.log = log
         if log is None:
             self.log = logModule(self.__class__.__name__)
             self.log.setLevel( self.log.INFO )
         self.session = session
-        self.playbackTool = playerTool["tool"]
-        for cmd in playerTool["prerequisites"]:
-            self.session.write(cmd)
+        self.playerProfile = ConfigRead(self.playerConfig, platform)
+        self.player = self.playerProfile.get(player)
+
+        if self.player.get("prerequisites"):
+            for cmd in self.player.get("prerequisites"):
+                self.session.write(cmd)
 
-    def play(self, streamFile:str):
+    def play(self, streamFile:str, mixer_input:str = "primary"):
```

Comment:
your mixing the stream play requirements and the player configuration requirements.

you should have a new function for configuration, and add more if required.

But really you should be using enums [https://docs.python.org/3/library/enum.html](https://docs.python.org/3/library/enum.html)

from enum import Enum

```pytnon
# class syntax
class MixerInputTypes(Enum):
    MIXER_INPUT_PRIMARY= 1
    MIXER_INPUT_SECONDARY = 2
...
def setMixerInput( self, mixer_input:MixerInputTypes )
````
In init, you should set the MixerInput(MIXER_INPUT_PRIMARY), then if anyone wants to change it after init they call the function again.

Reply:
done